corehq/apps/api/resources/v0_5.py

```diff
@@ -364,16 +380,31 @@ def _update(self, bundle, user_change_logger=None):
                 continue
             if getattr(bundle.obj, key, None) != value:
                 if key == 'phone_numbers':
+                    old_phone_numbers = set(bundle.obj.phone_numbers)
+                    new_phone_numbers = set()
                     bundle.obj.phone_numbers = []
                     for idx, phone_number in enumerate(bundle.data.get('phone_numbers', [])):
                         formatted_phone_number = strip_plus(phone_number)
-                        if user_change_logger and formatted_phone_number not in bundle.obj.phone_numbers:
-                            user_change_logger.add_change_message(
-                                UserChangeMessage.phone_number_added(formatted_phone_number))
+                        new_phone_numbers.add(formatted_phone_number)
                         bundle.obj.add_phone_number(formatted_phone_number)
                         if idx == 0:
                             bundle.obj.set_default_phone_number(formatted_phone_number)
                         should_save = True
+
+                    if user_change_logger:
```

Comment:
These two blocks are the same - can you pull them out into a helper function?  That'd also help keep these methods a bit shorter

Reply:
I am expecting `WebUserResource` to get removed completely in https://dimagi-dev.atlassian.net/browse/SAAS-12603, so good to leave it as it is?